### PR TITLE
statistics: move TableStatsFromStorage out of handle

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Issue Number: ref #40791 ref #40624

Problem Summary:

### What is changed and how it works?

Move `TableStatsFromStorage` out of `statistics/handle`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/interact_with_storage.go [hunk 1]
```diff
@@ -16,13 +16,23 @@ package statistics
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"go.uber.org/zap"
```
Comment (Reminiscent) [inline statistics/interact_with_storage.go:21]: format

File: statistics/handle/handle.go [hunk 2]
```diff
@@ -1420,97 +1396,28 @@ func (h *Handle) TableStatsFromStorage(tableInfo *model.TableInfo, physicalID in
...
-						return nil, err
-					}
-				}
-			} else {
-				item.StringVals = statsStr
-			}
-			table.ExtendedStats.Stats[name] = item
+	for histID, col := range statsTbl.Columns {
+		if statistics.IsAnalyzed(col.Flag) {
+			h.mu.rateMap.clear(physicalID, histID, false)
```
Comment (Reminiscent) [inline statistics/handle/handle.go:1417]: Why move this part to here. Not in the origin function `columnStatsFromStorage` and `indexStatsFromStorage`

**Review by Reminiscent — APPROVED** [review]
Rest LGTM

**Review by Yisaer — APPROVED** [review]

File: statistics/handle/handle.go (hunk 2 above)
Comment (xuyifangreeneyes) [reply statistics/handle/handle.go:1417]: Because we need `h.mu.rateMap` for the part.

**Pull request merged**